### DISCUSSION TITLE
Suppress Version

### DISCUSSION THREAD

**Discussion by bh** [dani-garcia/vaultwarden#2458] [Q&A]
Is there any option to suppress the Vaultwarden version?

`© 2022, Bitwarden Inc. (Powered by Vaultwarden)
Version 2.28.0 `

**Comment by BlackDex** [comment]
Build the web-vault your self.

The web-vault distributed with our build need to have that.

May i ask why you want that btw?

**Reply by bh** [reply]
For security reasons. Something like this here: http://nginx.org/en/docs/http/ngx_http_core_module.html#server_tokens

**Reply by BlackDex** [reply]
Well, no we can't make a toggle for this.
Also that version is the web-vault version, not the server version.
And if someone want's to know if it is Vaultwarden or the original Bitwarden server there are other means on determining that.

So if you really do not want it, you need to build the web-vault your self and replace it.

**Reply by brasilikum** [reply]
@BlackDex, I would like to understand the decision to not make a toggle for this better.

I cite from the [vaultwarden hardening-guide](https://github.com/dani-garcia/vaultwarden/wiki/Hardening-Guide) re subdir:
> Some may argue that this is [security through obscurity](https://en.wikipedia.org/wiki/Security_through_obscurity), but it's actually [defense in depth](https://en.wikipedia.org/wiki/Defense_in_depth_(computing)) -- the secrecy of the subdir is just an extra layer of security, and not intended to be the primary means of security (which is still the strength of a user's master password).

Doesn't the same apply to hiding the version?
Of course, it's the version of the web-vault, not the server, but it's likely attackers can conclude from one to the other, no?

**Reply by BlackDex** [reply]
@brasilikum mostly because the web-vault is not dynamically configurable from the server side for items likes that.
Also, though the container images we build have a version of the web-vault we determine our self. People can still override this. Or some do not use the container at all, and might use a different version of the web-vault etc... etc...

So there are many variables which makes it hard, but the main one is that there are no configuration flags for toggling that provided from upstream Bitwarden. 

And if you really want to remove it, i suggest to use @jjlin solution below.

**Comment by jjlin** [comment]
While it doesn't provide any real security, if you really want to do it without rebuilding the web vault, a quick and dirty way is to patch the web vault app on startup by running something like
```sh
sed -i 's/"versionNumber"/"versionNumberDisabled"/g' /web-vault/app/main.*.js
```
This should be put in a [container init script](https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container#customizing-container-startup).
